### ISSUE TITLE
`rke config` generates inconsistent `cluster.yml`

### ISSUE THREAD

**Issue by HectorB-2020** [rancher/rke#3295]
**RKE version:**
1.4.7

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
Ubuntu 20.04 LTS

**Steps to Reproduce:**
1. Run `rke config`, provide answers.
2. Give k8s version as one of answers, e.g. `v1.24.15-rancher1`.
3. Try running `rke up`.
4. Watch error like
    rke [controlPlane] Failed to bring up Control Plane: [Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host couldn't init admission plugin PodSecurity

**Explanation:**
When `rke config` seeks for responses, it puts k8s version only into `system_images`:
```yaml
system_images:
  etcd: rancher/mirrored-coreos-etcd:v3.5.4
  ...
  coredns_autoscaler: rancher/mirrored-cluster-proportional-autoscaler:1.8.5
  nodelocal: rancher/mirrored-k8s-dns-node-cache:1.21.1
  kubernetes: rancher/hyperkube:v1.24.15-rancher1     ## <----
  flannel: rancher/mirrored-coreos-flannel:v0.15.1
  flannel_cni: rancher/flannel-cni:v0.3.0-rancher6
  ...
```
Parameter `kubernetes_version` is left empty after `rke config`:
`kubernetes_version: ""`

Thanks to old issue rancher/rke#3134 I found a mitigation - we needed to explicitly define
`kubernetes_version: "v1.24.15-rancher1"`

Funny thing, before playing with `rke` and `cluster.yml` I had perused all documentation which said that information defined in `system_images` prevails. I certainly had seen empty `kubernetes_version: ""` but never though it made any sense for my configuration.

So bottom line, it would be nice to have ready-to-use `cluster.yml` after providing answers to `rke config` which include k8s version.

----

BTW, I hope you'll also be able to look into other issue with `system_images`:
* https://github.com/rancher/rke/issues/3294

**Comment by HectorB-2020** [comment]
I wonder if this is relevant. The following command explicitly specifies version `1.24.15` but its logs say that it believes that version `1.26.6` was chosen.
```bash
rke config --system-images -v v1.24.15-rancher1
```
`level=info msg="Generating images list for version [v1.26.6-rancher1-1]`

**Comment by HectorB-2020** [comment]
And I guess something related was discussed in https://github.com/rancher/rke/issues/2532#issuecomment-832896860

**Comment by jiaqiluo** [comment]
> I wonder if this is relevant. The following command explicitly specifies version `1.24.15` but its logs say that it believes that version `1.26.6` was chosen.
> 
> ```shell
> rke config --system-images -v v1.24.15-rancher1
> ```
> 
> `level=info msg="Generating images list for version [v1.26.6-rancher1-1]`

 @HectorB-2020, in this case, the default version, which is v1.26.6-rancher1-1,  will be used. 
https://github.com/rancher/rke/blob/5a8cacb962809382906100bcb8e797e9c8144f68/cmd/config.go#L296-L310

To check the default version, run `rke config -l`
```
> rke config -l   
v1.26.6-rancher1-1

>  rke -v       
rke version v1.4.7
```

**Comment by HectorB-2020** [comment]
@jiaqiluo I hope you can help a bit more.
There are two places where Kubernetes version is specified in `cluster.yml`.
```yaml
kubernetes_version: "v1.24.15-rancher1"
...
system_images:
  kubernetes: rancher/hyperkube:v1.24.15-rancher1
```
Both places expect k8s version without the tailing digit, don't they? Am I right?
Yet `rke config -l` lists k8s versions with the last digits. This is confusing.
So I wonder how these two places relate to each other in the configuration.

**Comment by jiaqiluo** [comment]
@HectorB-2020,

To set the k8s version of your cluster, you just need to set `kubernetes_version` in the cluster.yml file.  the `system_images` section can be left for RKE to fill internally.

You need the trailing digit.  The output of `rke config -l` ( or `rke config -l -a` ) are the list of exact values you should set for `kubernetes_version`

**Comment by HectorB-2020** [comment]
Thanks very much again @jiaqiluo!
I certainly shall follow this approach with setting `kubernetes_version:` with trailing digits like `v1.24.15-rancher1-1`.

However this is still confusing. The way `rke config` asks questions it's obvious that it seeks for exact container tag. With my choice this should be
```yaml
system_images:
  kubernetes: rancher/hyperkube:v1.24.15-rancher1
```
But even if I run this command explicitly setting different k8s version 
```bash
rke config -v v1.24.15-rancher1-1
```
it still suggests `rancher/hyperkube:v1.26.6-rancher1`. 

In other words, my understanding is that the `rke` tool never asks for `kubernetes_version:` and disregards setting versions with the command line parameters.

**Comment by jiaqiluo** [comment]
@HectorB-2020, yeah, I agree that it looks like a bug.  Please use the `kubernetes_version:` for setting k8s versions for now.

**Comment by HectorB-2020** [comment]
Hello @jiaqiluo!
As we see `rke` v1.4.8 has been released. And it introduced newer versions of Kubernetes:
* v1.24.16-rancher1-1
* v1.25.12-rancher1-1
* v1.26.7-rancher1-1

Does this mean that previous Kubernetes versions cannot be used anymore?
I'm concerned since our environment is purely air-gapped and my `cluster.yaml` contains `v1.24.15-rancher1-1` as discussed above. Obviously that image had been pushed to our private registry.
Shall we revise the list of images to be pulled anew, compressed and sent to our private registry?

**Comment by jiaqiluo** [comment]
@HectorB-2020 , you can still use the old versions. by setting the `kubernetes_version` in the cluster.yml file, they are just hidden in favor of the new ones.

**Comment by HectorB-2020** [comment]
Awesome, thanks @jiaqiluo!
